### PR TITLE
updated setupWeb3 and exported for use with new process

### PR DESCRIPTION
**Pull request by W3stside**
Finishes DX-648

Enables window.etheruem.enable in load

### CODE REVIEW

**Comment by ghost** [comment]
Travis automatic deployment:
 https://pr422--dxreact.review.gnosisdev.com

**Comment by ghost** [comment]
Travis automatic deployment:
 https://pr422--dxreact.review.gnosisdev.com

File: src/api/web3Provider.ts [hunk 1]
```diff
@@ -3,9 +3,31 @@ import { windowLoaded, promisify } from 'utils'
...
 import Web3 from 'web3'
 
-const setupWeb3 = async (provider: Provider) => {
+export const setupWeb3 = async (provider?: Provider) => {
   await windowLoaded
 
+  if (!provider) {
+    let providerWeb3: any
+    if (window.ethereum) {
+      providerWeb3 = window.web3 = new Web3(window.ethereum)
```
Comment (Velenir) [inline src/api/web3Provider.ts]: no need to `let providerWeb3`, do `const` below

**Review by Velenir — APPROVED** [review]

File: src/api/web3Provider.ts (hunk 1 above)
Comment (W3stside) [reply src/api/web3Provider.ts]: yep, legacy from shitty testing

**Pull request merged**